Code/Mantid/Framework/RemoteJobManagers/src/LSFJobManager.cpp

```diff
@@ -0,0 +1,1257 @@
+      session.setMethod(method);
+    }
+  }
+  return session.sendRequest(url, rss);
+}
+
+/**
+ * Fills in a table workspace with job status information from an LSC
+ * PAC response in ~xml format. Assumes that the workspace passed is
+ * empty and ready to be filled. This guarantees that a non-null (I)
+ * table workspace object is returned.
+ *
+ * @param resp Body of an HHTP response to a status query
+ * @param jobIDFilter ID of one job (empty string immplies all jobs)
+ *
+ * @return vector with status and related information for all jobs reported by
+ * the server.
+ */
+std::vector<Mantid::API::IRemoteJobManager::RemoteJobInfo>
+LSFJobManager::genOutputStatusInfo(const std::string &resp,
+                                   const std::string &jobIDFilter) const {
+  Poco::XML::DOMParser parser;
+  Poco::AutoPtr<Poco::XML::Document> doc;
+  try {
+    doc = parser.parseString(resp);
+  } catch (Poco::Exception &e) {
+    throw std::runtime_error("Unable to parse response in XML format: " +
+                             e.displayText());
+  } catch (std::exception &e) {
+    throw std::runtime_error("Unable to parse response in XML format: " +
+                             std::string(e.what()));
+  }
+
+  Poco::XML::Element *pRootElem = doc->documentElement();
+  if (!pRootElem || !pRootElem->hasChildNodes()) {
+    g_log.error("XML response from compute resouce contains no root element.");
+    throw std::runtime_error("No root element was found in XML response, "
+                             "cannot parse it.");
+  }
+
+  Poco::AutoPtr<Poco::XML::NodeList> jobs =
+      pRootElem->getElementsByTagName("Job");
+  if (!jobs) {
+    g_log.error("XML response from compute resouce contains no root element.");
+    throw std::runtime_error("No root element was found in XML response, "
+                             "cannot parse it.");
+  }
+
+  size_t n = jobs->length();
+  if (0 == jobs->length()) {
+    g_log.notice() << "Got information about 0 jobs. You may not have any jobs "
+                      "currently running on the compute resource. The output "
+                      "workspace will not "
+                      "have any rows/information";
+  }
+
+  // This is the information that is usually available for running/recently
+  // run jobs: id, name, status, command line
+  std::vector<IRemoteJobManager::RemoteJobInfo> info;
+  for (size_t i = 0; i < n; i++) {
+    Poco::XML::Element *el = static_cast<Poco::XML::Element *>(
+        jobs->item(static_cast<unsigned long>(i)));
+    if (!el)
+      throw std::runtime_error("Error while trying to parse job with index " +
+                               boost::lexical_cast<std::string>(i) +
+                               "could not produce a complete table workspace.");
+
+    info.push_back(RemoteJobInfo());
+
+    Poco::XML::Element *id = el->getChildElement("id");
+    if (id) {
+      const std::string &IdStr = id->innerText().c_str();
```

Comment:
If innerText() return std::string there is no need to cast to a c-string and create a temporary.